tokio-util/src/sync/mpsc.rs

```diff
@@ -64,13 +64,45 @@ async fn make_acquire_future<T>(
     }
 }
 
-impl<T: Send + 'static> PollSender<T> {
+#[derive(Debug)]
+// TODO: This should be replace with a type_alias_impl_trait to eliminate `'static` and all the transmutes
+struct PollSenderFuture<T>(ReusableBoxFuture<'static, Result<OwnedPermit<T>, PollSendError<T>>>);
+
+impl<T> From<ReusableBoxFuture<'static, Result<OwnedPermit<T>, PollSendError<T>>>>
+    for PollSenderFuture<T>
+{
+    fn from(v: ReusableBoxFuture<'static, Result<OwnedPermit<T>, PollSendError<T>>>) -> Self {
+        Self(v)
+    }
+}
+
+impl<T: Send> From<Option<Sender<T>>> for PollSenderFuture<T> {
+    fn from(data: Option<Sender<T>>) -> Self {
+        let v = ReusableBoxFuture::<'_, Result<OwnedPermit<T>, PollSendError<T>>>::new(
+            make_acquire_future(data),
+        );
+        // This is safe because we're just transmuting the `'static` bound on the Box<dyn> to get around storage issue
+        unsafe { mem::transmute(v) }
```

Comment:
Here you are not just transmuting the lifetime, you are also transmuting it into a `PollSenderFuture`. That's not valid unless it is `#[repr(transparent)]`. Please construct the `PollSenderFuture` separately after the transmute, and please write out the generic types on the transmute to make it clear what the from and to types are.